tap_salesforce/salesforce/__init__.py

```diff
@@ -381,13 +387,25 @@ def _get_selected_properties(self, catalog_entry):
                                             self.select_fields_by_default)]
 
 
-    def get_start_date(self, state, catalog_entry):
+    def get_start_date(self, state, catalog_entry, with_lookback=False):
+        """
+            if 'with_lookback' is False, then return start date or state file date
+            else subtract lookback window from the state file date and return
+                if no state file bookmark is found, do not subtract lookback window from the start date
+        """
         catalog_metadata = metadata.to_map(catalog_entry['metadata'])
         replication_key = catalog_metadata.get((), {}).get('replication-key')
 
-        return (singer.get_bookmark(state,
-                                    catalog_entry['tap_stream_id'],
-                                    replication_key) or self.default_start_date)
+        sync_start_date = singer.get_bookmark(state, catalog_entry['tap_stream_id'], replication_key) or self.default_start_date
+        # return bookmark from the state or start date if 'with_lookback' is False
+        if not with_lookback:
+            return sync_start_date
+
+        # if the state contains a bookmark, subtract the lookback window from the bookmark
+        if singer.get_bookmark(state, catalog_entry['tap_stream_id'], replication_key):
+            sync_start_date = singer_utils.strftime(singer_utils.strptime_with_tz(sync_start_date) - datetime.timedelta(seconds=self.lookback_window))
+
+        return sync_start_date
```

Comment:
Update this implementation to get the tests in `tests/unittests/test_lookback_window.py` passing

Reply:
One helpful thing might be helpful is that `self.lookback_window` is an integer or None, so we can use its value or fallback to zero

Reply:
Thanks. Updated the code to only subtract lookback if the bookmark is present in the state and lookback is not `None`.